crates/polars-time/src/upsample.rs

```diff
@@ -130,6 +137,29 @@ fn upsample_impl(
         out.try_apply(index_column, |s| s.cast(&DataType::Date))
             .unwrap();
         Ok(out)
+    } else if matches!(s.dtype(), DataType::Int32) {
+        let mut df = source.clone();
+
+        df.try_apply(index_column, |s| {
+            s.cast(&DataType::Int64)
+                .unwrap()
+                .cast(&DataType::Datetime(TimeUnit::Nanoseconds, None))
+        })
+        .unwrap();
```

Comment:
if you're unwrapping the result of `cast` then there's not much use of using `try_apply` - I'd suggest just using `apply` and `unwrap` inside it

Reply:
Fixed.